nomenclature/error/variable.py

```diff
@@ -0,0 +1,17 @@
+from pydantic import PydanticValueError
+
+
+class VariableRenameArgError(PydanticValueError):
```

Comment:
Since both of the custom errors here and not used in a pydantic class but in a regular one, sub-classing pydantic errors feels a bit wrong to me. It perfectly works but to me it suggests that they are used inside a pydantic class which is not the case.
Something like this should work just as well:
```suggestion
class VariableRenameArgError(ValueError):
    def __init__(self, args, variable, file):
        super().__init__(f"Using attribute 'region-aggregation' and arguments {args} not supported, occurred in variable '{variable}' (file: {file})")
```
The main appeal of pydantic Errors is that all errors raised during validation are collected and only raised in the very end as a single container error containing a list of all individual errors. In our case a configuration like this:

```yaml
- Price|Carbon:
    definition: Price of carbon
    unit: USD/t CO2
    weight: Emissions|CO2
    region-aggregation:
      - Price|Carbon (Max):
          method: max
```

should raise both errors in this module. Since `DataStructureDefinition` is not a pydantic class I would prefer to use errors sub-classed from 'normal' python exceptions.